src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -31,7 +32,7 @@ public partial class Regex : ISerializable
         protected internal int capsize;                       // the size of the capture array
 
         internal WeakReference<RegexReplacement?>? _replref;  // cached parsed replacement pattern
-        private ExclusiveReference? _runnerref;               // cached runner
+        private volatile RegexRunner? _runner;                // cached runner
```

Comment:
Do you still need `volatile` here since you are using `Interlocked.Exchange`?

Reply:
The `volatile` won't be relevant for the `Interlocked.Exchange`, but it's still relevant for the release in https://github.com/dotnet/runtime/pull/542/files/3dac3dae8db5be5254da9317f0db9756f43572e6#diff-342612433ebf6c26c55fcd0d4d10b9baR449.  From a correctness perspective, it's probably not necessary, but I felt better having it.

Reply:
Sounds good to me.